src/components/storage/aws.ts

```diff
@@ -148,35 +128,13 @@ export class AwsEventLogStorage implements EventLogStorage {
             Bucket: this.bucketName,
             ContentType: contentType
         }
-    
-        await new Promise((resolve, reject) => {
-            this._s3.putObject(params, (err, data) => {
-                err ? reject(err) : resolve()
-            })
-        })
-    }
 
-    async _getObject({key, type} : {key : string, type? : 'csv'}) {
-        const params = {
-            Bucket: this.bucketName,
-            Key: key,
+        try {
+            const headCode = this._s3.putObject(params).promise()
```

Comment:
From what I understand `headCode` is going to be set to type `Promise` then this method will always immediately return `true` regardless of the outcome of the Promise. The `catch` scope will never be entered as you would need an `await` to wait until the Promise resolves or rejects. 
Same issue with both the classes, although the `_getObject` methods seem to wait for the Promise's outcome ok (although I would avoid assigning `headCode` variable then never using it - you could set up tslint to warn you about unused assignments).

In general I would avoid having these get/putObject methods return booleans depending on the outcome of the Promises. They already return a Promise (as they're async methods), so you could just replace your `return false` calls with `throw someError` (translates to `Promise.reject()` outside of asyncawait) then the callers can simply use `try...catch`/`.then().catch()` and decide on how to act (rather than waiting on the Promise to resolve to either `true` or `false` and the possibility of handling a rejection that is semantically different to `false`).